packages/react/src/transactions/adapters/responders/__tests__/UpdateProfileImageResponder.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { faker } from '@faker-js/faker';
+import { ProfileFieldsFragment, ProfileFieldsFragmentDoc } from '@lens-protocol/api-bindings';
+import {
+  createMockApolloCache,
+  mockProfileFieldsFragment,
+} from '@lens-protocol/api-bindings/mocks';
+import {
+  mockBroadcastedTransactionData,
+  mockUpdateOffChainProfileImageRequest,
+} from '@lens-protocol/domain/mocks';
+
+import { UpdateProfileImageResponder } from '../UpdateProfileImageResponder';
+
+function setupTestScenario({ profile }: { profile: ProfileFieldsFragment }) {
+  const apolloCache = createMockApolloCache();
+
+  apolloCache.writeFragment({
+    id: apolloCache.identify(profile),
+    fragment: ProfileFieldsFragmentDoc,
+    fragmentName: 'ProfileFields',
+    data: profile,
+  });
+
+  const responder = new UpdateProfileImageResponder(apolloCache);
+
+  return {
+    responder,
+
+    get profileFromCache() {
+      return apolloCache.readFragment({
+        id: apolloCache.identify(profile),
+        fragment: ProfileFieldsFragmentDoc,
+        fragmentName: 'ProfileFields',
+      });
+    },
+  };
+}
+
+describe(`Given an instance of the ${UpdateProfileImageResponder.name}`, () => {
+  const profile = mockProfileFieldsFragment();
+
+  describe(`when "${UpdateProfileImageResponder.prototype.commit.name}" method is invoked`, () => {
+    const newImageUrl = faker.image.imageUrl();
+
+    const transactionData = mockBroadcastedTransactionData({
+      request: mockUpdateOffChainProfileImageRequest({
+        profileId: profile.id,
+        url: newImageUrl,
+      }),
+    });
+
+    it(`should update profile picture to the latest one`, async () => {
+      const scenario = setupTestScenario({ profile });
+
+      await scenario.responder.commit(transactionData);
+
+      expect(scenario.profileFromCache).toEqual(profile);
```

Comment:
it appears this test is a little bit broken.
`commit` should modify the profile `picture`, then how it's possible that `scenario.profileFromCache === profile` after updating the cache with a new picture? 

it appears that the `arrange` part of the unit test is wrong as we don't set up a different profile image than the one generated by `faker.image.imageUrl()`.
I'm surprised that `faker.image.imageUrl` returns the same URL every time, that's why the test is passing.
When creating a profile on line `40` we do need to provide some fake `url` which we would then try to overwrite with a new fake url (i suggest to not use `faker.image.imageUrl` at all given it doesn't create random url)

Reply:
wow, good catch, I didn't realize that faker is always returning the same image URL. I will fix that.

Reply:
so the 4th argument is a randomize flag :) I fixed the test
```
faker.image.imageUrl(1234, 2345, 'cat', true) // 'https://loremflickr.com/1234/2345/cat?lock=6849'
```

Reply:
wat? Faker returns the same image URL?